pkg/executor/batch_checker.go

```diff
@@ -48,6 +49,12 @@ type toBeCheckedRow struct {
 	uniqueKeys []*keyValueWithDupInfo
 	// t is the table or partition this row belongs to.
 	t       table.Table
+	// oldRowTable only used for global index.
+	// `insert ... on duplicate key update` inserts a row, but the global index uk may have existed.
+	// the partition id in the global index value is the real tableID for old row.
+	// use oldRowTable to temporarily store the table this old row belongs to.
+	// `TestGlobalIndexForClusteredIndexTable` is a concrete case for the scenario.
+	oldRowTable table.Table
```

Comment:
> I don't really understand why this is needed, `t` should be the the same as `oldRowTable`, since it should be updated to the partition [here](https://github.com/pingcap/tidb/blob/4abe19251ddb38bf83cb0b2a16e0c77228f772df/pkg/executor/batch_checker.go#L101).

For partitioned table, `t` is partition the new record row belongs to. But the coresponding global index row may exist and it belongs to old partition. The old partition id exists in the global index value.

Reply:
Is it only used for `REPLACE`? or is it different from `t` for any other statement?